File: jurij_veresciaka/U1/code/php/includes/model/AngleCalculator.php
<?php

require_once dirname(__FILE__) . DIRECTORY_SEPARATOR . "Clock.php";

class AngleCalculator
{

    ///////////////////////////////////////////////
    //VARIABLES                                  //
    ///////////////////////////////////////////////

    private $clock;
    private $minutes_arrow_minutes_multiplicator;
    private $minutes_arrow_seconds_multiplicator;

    ///////////////////////////////////////////////
    //CONSTRUCTOR                                //
    ///////////////////////////////////////////////

    function __construct($clock)
    {
        $this->clock = $clock;

        $this->minutes_arrow_minutes_multiplicator = 6.0;
        $this->minutes_arrow_seconds_multiplicator = 0.1;

        $this->hours_arrow_hours_multiplicator = 30.0;
        $this->hours_arrow_minutes_multiplicator = 0.5;
        $this->hours_arrow_seconds_multiplicator = 30 / 3600;
    }

    ///////////////////////////////////////////////
    //CUSTOM GETTERS                             //
    ///////////////////////////////////////////////

    public function getMinutesArrowAngle()
    {
        $angle = ($this->clock->getMinutes() * $this->minutes_arrow_minutes_multiplicator);
        $angle += ($this->clock->getSeconds() * $this->minutes_arrow_seconds_multiplicator);

        return $angle;
    }

    public function getHoursArrowAngle()
    {
        $angle = ($this->clock->getHours12Format() * $this->hours_arrow_hours_multiplicator);
        $angle += ($this->clock->getMinutes() * $this->hours_arrow_minutes_multiplicator);
        $angle += ($this->clock->getSeconds() * $this->hours_arrow_seconds_multiplicator);

        return $angle;
    }

    public function getMinimumAngleBetweenClockArrowsFloat()
    {
        $angle = abs($this->getMinutesArrowAngle() - $this->getHoursArrowAngle());

        if ($angle >= 180.0) {
            $angle = 360.0 - $angle;
        }

        return $angle;
    }

    public function getMinimumAngleBetweenClockArrowsString()
    {
        $angle = strval($this->getMinimumAngleBetweenClockArrowsFloat() . "&deg;");

        return $angle;
    }

    public function getMaximumAngleBetweenClockArrowsFloat()
    {
        $angle = abs($this->getMinutesArrowAngle() - $this->getHoursArrowAngle());

        if ($angle < 180.0) {
            $angle = 360.0 - $angle;
        }

        return $angle;
    }

    public function getMaximumAngleBetweenClockArrowsString()
    {
        $angle = strval($this->getMaximumAngleBetweenClockArrowsFloat() . "&deg;");

        return $angle;
    }

    ///////////////////////////////////////////////
    //ACCESSORS                                  //
    ///////////////////////////////////////////////

    /**
     * @param mixed $clock
     */
    public function setClock($clock)
    {
        $this->clock = $clock;
    }

    /**
     * @return mixed
     */
    public function getClock()
    {
        return $this->clock;
    }
}

File: jurij_veresciaka/U1/code/javase/src/model/AngleCalculator.java
package model;

public class AngleCalculator {

    ///////////////////////////////////////////////
    //VARIABLES                                  //
    ///////////////////////////////////////////////
	
	private Clock clock;
	
    private float minutesArrowMinutesMultiplicator = 6.0f;
    private float minutesArrowSecondsMultiplicator = 0.1f;
    
    private float hoursArrowHoursMultiplicator = 30.0f;
    private float hoursArrowMinutesMultiplicator = 0.5f;
    private float hoursArrowSecondsMultiplicator = 30f / 3600f;
    
	///////////////////////////////////////////////
    //CONSTRUCTOR                                //
    ///////////////////////////////////////////////
    
	public AngleCalculator(Clock clock) {
		this.clock = clock;
	}
	
    ///////////////////////////////////////////////
    //CUSTOM GETTERS                             //
    ///////////////////////////////////////////////

    public float getMinutesArrowAngle()
    {
    	float angle = 0f;
    	
    	angle = this.clock.getMinutes() * this.minutesArrowMinutesMultiplicator;
    	angle += this.clock.getSeconds() * this.minutesArrowSecondsMultiplicator;

        return angle;
    }
    
    public float getHoursArrowAngle()
    {
    	float angle = 0f;
    	
    	angle = this.clock.getHours24Format() * this.hoursArrowHoursMultiplicator;
    	angle += this.clock.getMinutes() * this.hoursArrowMinutesMultiplicator;
    	angle += this.clock.getSeconds() * this.hoursArrowSecondsMultiplicator;

        return angle;
    }
    
    public float getMinimumAngleBetweenClockArrowsFloat()
    {
        float angle = Math.abs(this.getMinutesArrowAngle() - this.getHoursArrowAngle());

        if (angle >= 180.0) {
        	angle = 360.0f - angle;
        }

        return angle;
    }
    
    public String getMinimumAngleBetweenClockArrowsString()
    {
        String angle = Float.toString(this.getMinimumAngleBetweenClockArrowsFloat());
        angle += "\u00B0";

        return angle;
    }
    
    public float getMaximumAngleBetweenClockArrowsFloat()
    {
        float angle = Math.abs(this.getMinutesArrowAngle() - this.getHoursArrowAngle());

        if (angle < 180.0) {
        	angle = 360.0f - angle;
        }

        return angle;
    }
    
    public String getMaximumAngleBetweenClockArrowsString()
    {
        String angle = Float.toString(this.getMaximumAngleBetweenClockArrowsFloat());
        angle += "\u00B0";

        return angle;
    }
}
